apps : Add --gc-sections flag in application Makefile

drashti304 opened the pull request:
This patch adds --gc-sections flag in application makefile.
this tells linker to remove unused sections from the code,
which results in reducing code size.

seinfra commented:
Target : [156150c85535df4c6b51af6d014385ced627d754] - Code Rule Check OK.

seinfra commented:
Target : [156150c85535df4c6b51af6d014385ced627d754] - Code Rule Check (C++) OK.

yashwanth686007 commented on apps/examples/elf/micomapp/Makefile:
```diff
@@ -92,6 +92,7 @@ COMPRESSION_TYPE = 0
 BLOCK_SIZE = 0
 endif
 
+LDELFFLAGS += --gc-sections
```
Is it possible to add this in AppLibs.mk so that redundancy can be reduced.

drashti304 replied:
yeah, done

seinfra commented:
Target : [b0cc734e5db31c1458d3c7059a0fa1c36852eb81] - Code Rule Check OK.

seinfra commented:
Target : [b0cc734e5db31c1458d3c7059a0fa1c36852eb81] - Code Rule Check (C++) OK.

seinfra commented:
Target : [f2b7f2c07b129b6780f92b91e0bd196ea662ddfe] - Code Rule Check OK.

seinfra commented:
Target : [f2b7f2c07b129b6780f92b91e0bd196ea662ddfe] - Code Rule Check (C++) OK.

sijupunnnoose commented on apps/AppLibs.mk:
```diff
@@ -60,4 +60,5 @@ LDLIBS_LIST = $(patsubst %.a, %, $(patsubst lib%,-l%,$(LINKLIBS)))
 LDLIBS = $(patsubst -lapps%, %, $(LDLIBS_LIST))
```
Can you change the commit message as apps/examples/elf -> apps ??

drashti304 commented:
> Can you change the commit message as apps/examples/elf -> apps ??

Updated commit message from apps/examples/elf -> apps/AppLibs.mk

seinfra commented:
Target : [dc785186812bb7235dcc2fc4f2aeb808476a491a] - Code Rule Check OK.

seinfra commented:
Target : [dc785186812bb7235dcc2fc4f2aeb808476a491a] - Code Rule Check (C++) OK.

sunghan-chang commented:
@drashti304 You tested and said, there is no difference with `--gc-sections`. Why do you make this?

drashti304 commented:
> @drashti304 You tested and said, there is no difference with `--gc-sections`. Why do you make this?

Hello @sunghan-chang,
Sorry I made a mistake in testing NOOPT with --gc-sections flag. 
there is a reduction in size with "--gc-sections" when NOOPT is there, from 44945 -> 43373.
Actually I added --gc-sections in micom app, and later verified the size of wifiapp. So I thought there is no change in size.

So I added this flag in this PR
